### PR TITLE
Update Search Config to Override Max Guess Total

### PR DESCRIPTION
**Pull request by schmidtkevina**
Updated SearchConfig.java and SearchConfigImpl.java to allow for overring the max guess total value via a property from the Results component dialog.

### CODE REVIEW

File: core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/SearchConfigImpl.java [hunk 1]
```diff
@@ -119,12 +121,23 @@ public String getGuessTotal() {
 
         try {
             int guessTotalAsNumber = Integer.parseInt(guessTotal);
-            return (guessTotalAsNumber < 1 || guessTotalAsNumber > MAX_GUESS_TOTAL) ? DEFAULT_GUESS_TOTAL : String.valueOf(guessTotalAsNumber);
+            if(isOverideGuessTotal() && guessTotalAsNumber > 1) {
+                return guessTotal;
+            }
+            else {
+                return (guessTotalAsNumber < 1 || guessTotalAsNumber > MAX_GUESS_TOTAL) ? DEFAULT_GUESS_TOTAL : String.valueOf(guessTotalAsNumber);
+            }
```
Comment (davidjgonzalez) [inline core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/SearchConfigImpl.java:129]:
```suggestion
            if (isOverideGuessTotal() && guessTotalAsNumber >= -1) {
                 // Override allowed; Allows any values -1, 0 or greater if the override is set, as -1 and 0 set "no limit"
                return String.valueOf(guessTotalAsNumber);
            } else if (guessTotalAsNumber < 1 || guessTotalAsNumber > MAX_GUESS_TOTAL) {
               // Override NOT allowed, and value outside allowed bounds so use default
               return DEFAULT_GUESS_TOTAL;
            } else {
               // Override NOT allowed, but value is within the allowed bounds
               return String.valueOf(guessTotalAsNumber);
            }
```

File: core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/SearchConfigImpl.java [hunk 2]
```diff
@@ -119,12 +121,23 @@ public String getGuessTotal() {
...
         } catch (NumberFormatException e) {
             return DEFAULT_GUESS_TOTAL;
         }
     }
 
+    @Override 
+    public boolean isOverideGuessTotal() {
```
Comment (godanny86) [inline core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/SearchConfigImpl.java:140]:
```suggestion
    public boolean isOverrideGuessTotal() {
```

Update the spelling?

File: ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/search/results/_cq_dialog/.content.xml [hunk 3]
```diff
@@ -93,6 +93,14 @@
                                         text="Display help message as placeholder"
                                         value="100"/>
 
+                                    <override-guess-total
+                                        jcr:primaryType="nt:unstructured"
+                                        sling:resourceType="granite/ui/components/coral/foundation/form/checkbox"
+                                        fieldDescription="Select to override the default max guess total of 2000.  It's recommended to keep the max guess total at 2000 or less."
+                                        name="./overrideGuessTotal"
+                                        text="Override Default Max Guess Total (2000)"
```
Comment (davidjgonzalez) [inline ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/search/results/_cq_dialog/.content.xml:101]:
```suggestion
                                        fieldDescription="Allow for the override of the default max "guess total" of 2,000.  Beware, overriding the max guessTotal can result in performance degradation."
                                        name="./overrideGuessTotal"
                                        text="Allow override of Guess Total"
```

File: core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/SearchConfigImpl.java (hunk 2 above)
Comment (schmidtkevina) [reply core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/SearchConfigImpl.java]: I updated the spelling in the Interface and Impl as well.

**Comment by davidjgonzalez** [comment]
The alternative approach is setting this via OSGi config - as this can have serious system performance implications, which a person editing a component might not understand or appreciate. I do worry a little about ppl toggling this on and crashing out their AEM's with massive queries that can result in in-mem filtering (since QB does that depending on the query construction)

Just want to make sure this is thought through (there is a reason we had a max limit of 2000 set) .

**Comment by schmidtkevina** [comment]
I think the messaging is clear, that it can cause issues, if the number is over 2,000 and should be done with caution.

**Comment by davidjgonzalez** [comment]
What as the rationale of a dialog property vs a "harder to set" osgi property?

**Comment by schmidtkevina** [comment]
I know in our case, the authors wanted more control vs having have us set it for them as they have multiple search pages and add them often based on different groups.  Some search pages they want to show more than the max and some they don't, so each search results could be different.  Does that make sense?

**Pull request closed without merging**